chainermn/communicators/mpi_communicator_base.py

```diff
@@ -74,6 +105,12 @@ def alltoall(self, xs):
         invoke ``alltoall()``. This method relies on mpi4py fast communication
         optimized for numpy arrays, as well as ``send()`` and ``recv()``.
 
+        If ``xs`` is numpy array, the received data will also be allocated
+        as numpy array. If ``xs`` is cupy array, the received data will also
+        be cupy array. In the latter case, please be aware that
+        the CUDA current device is intended one.
+        (``https://docs-cupy.chainer.org/en/stable/tutorial/basic.html#current-device``)
```

Comment:
Nitpick English, but I think this would be more natural and precise.

> In the latter case, please be aware that the CUDA current device is intended one

to

> Additionaly, when an argument `xs` is cupy array, the returned array will be placed at [current device](https://docs-cupy.chainer.org/en/stable/tutorial/basic.html#current-device) regardless of which device the argument is placed at remote nodes.

In restructured text.
Docstrings in other methods also would have better be updated like this.